### PR TITLE
Add a --clarificationFile option

### PR DESCRIPTION
**Pull request by mikayla-maki**
While building out my company's automated licensing compliance code I attempted to use this project to implement automated monitoring for our project's compliance with the `--onlyAllow` flag. However, several of our dependencies (e.g. `@contentlayer/cli@0.3.0`) have an `UNKNOWN` license. I could not find a way to add package specific overrides with this project, so I decided to contribute to this project :). 

This PR adds a --clarificationFile option, modeled after the similar behavior in [`cargo-about`](https://embarkstudios.github.io/cargo-about/cli/generate/config.html#the-clarify-field-optional). This allows users to specify custom overrides for specific packages with missing or otherwise malformed fields. I also added tests for the feature, a `clarificationExample.json` file to show how it should work, and ensured that code coverage stayed within the project's limits. 

Let me know if there's anything else I can do for this PR, and thank you for maintaining such a vital project!

### CODE REVIEW

**Comment by mikayla-maki** [comment]
On further usage, I think it would be better to make the checksum field optional.

**Comment by RSeidelsohn** [comment]
Wow, this is by far the best PR I've seen so far for this project since I forked it from the original _license-checker_ - thank you so much!
I will have a look most likely tomorrow, but please bear with me if I can't do it before next week. I'm in winter holidays currently, using my free time for this project while the rest of my family is out for skying. Tomorrow will be the last day and it might be that I won't find even the little time necessary for having a look, even though I have the feeling that I can blindly trust your PR.
So getting back home will take us two days, but next week I will still be free. Should it be really urgent, please tell me so and I will put more pressure on my timetable.

Getting back to this PR soon & cheers,
Roman.

**Comment by mikayla-maki** [comment]
Hey Roman, 

Enjoy your holiday with your family, no rush to merge this PR! I have already pointed our internal builds at my fork branch so we're humming along just fine :)

Hope the skiing shreds 😄

**Comment by mikayla-maki** [comment]
In attempting to use this I found a need to snip out portions of the license from other embedded files. I have added the relevant functionality as well as some more documentation for the new features :)

File: tests/clarificationFile-test.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+        license_checker.on('exit', function(code) {
+            assert.equal(code, 0);
+            assert.equal(data.includes("MIT"), true)
+            assert.equal(data.includes("MY_IP"), true)
+            done();
+        });
+    })
+
+    it('should sniip the embedded license out of the README', function(done) {
```
Comment (RSeidelsohn) [inline tests/clarificationFile-test.js:83]: I guess this is a typo:

```suggestion
    it('should snip the embedded license out of the README', function(done) {
```

File: bin/license-checker-rseidelsohn [hunk 2]
```diff
@@ -47,6 +47,7 @@ const usageMessage = [
     '   --start [filepath] path of the initial json to look for',
     '   --summary output a summary of the license usage',
     '   --unknown report guessed licenses as unknown licenses.',
+    '   --clarificationsFile A file that describe the license clarifications for each package, see clarificationExample.json, any field available to the customFormat option can be clarified. Can also be used to specify a subregion of a package\'s license file (instead reading the entire file).',
```
Comment (RSeidelsohn) [inline bin/license-checker-rseidelsohn:50]: As the help message states _All options in alphabetical order_, I'd prefer having this line in line 23 instead. Sorry for being picky.

File: README.md [hunk 3]
```diff
@@ -183,6 +184,7 @@ before.
 -   `--start [filepath]` path of the initial json to look for
 -   `--summary` output a summary of the license usage',
 -   `--unknown` report guessed licenses as unknown licenses.
+-    `--clarificationsFile` A file that describe the license clarifications for each package, see clarificationExample.json, any field available to the customFormat option can be clarified. The clarifications file can also be used to specify a subregion of a package's license file (instead reading the entire file).
```
Comment (RSeidelsohn) [inline README.md:187]: As the headline states _All options in alphabetical order_, I'd prefer having this line in line 160 instead. Sorry for being picky.

**Review by RSeidelsohn — CHANGES_REQUESTED** [review]
Hi @mikayla-maki, once more thank you so much for your great contribution!
I have 3 really minor change requests of a purely aesthetic nature, I hope you don't mind.

I currently still struggle with setting up a proper Prettierjs and ESlint configuration that works seamlessly with the _Sublime Text 4_ setup I am using here, which will - should it work - demand single quotes rather than double quotes where applicable, but I hope this will not be an issue here for now.
What also irritates me is the fact that PRs come from foreign repositories, but that's a thing I have to learn, as it's the usual way things go in open source world. So maybe I handle things a bit strangely from time to time, but I hope my workflow and the project's documentation will improve over time.
That being said: I'm always open for questions and suggestions.

Cheers, Roman.

**Comment by mikayla-maki** [comment]
@RSeidelsohn Aesthetic changes implemented! Thank you for the feedback :D

**Review by RSeidelsohn — APPROVED** [review]

**Pull request merged**